openstack/objectstorage/v1/swauth/requests.go

```diff
@@ -0,0 +1,70 @@
+package swauth
+
+import "github.com/gophercloud/gophercloud"
+
+// AuthOptsBuilder describes struct types that can be accepted by the Auth call.
+// The AuthOpts struct in this package does.
+type AuthOptsBuilder interface {
+	ToAuthOptsMap() (map[string]string, error)
+}
+
+// AuthOpts specifies an authentication request.
+type AuthOpts struct {
+	// User is an Swauth-based username in username:tenant format.
+	User string `h:"X-Auth-User"`
+	// Key is a secret/password to authenticate the User with.
+	Key string `h:"X-Auth-Key"`
+}
+
+// ToAuthOptsMap formats an AuthOpts structure into a request body.
+func (opts AuthOpts) ToAuthOptsMap() (map[string]string, error) {
+	return gophercloud.BuildHeaders(opts)
+}
+
+// Authenticate performs an authentication request for a Swauth-based user.
+func Authenticate(c *gophercloud.ProviderClient, opts AuthOptsBuilder) (r GetAuthResult) {
```

Comment:
For consistency with the rest of the library, some options:
1. Rename function to `Auth` and result to `AuthResult`.
2. Rename result to `AuthenticateResult` and opts to `AuthenticateOpts`
3. Other suggestions